processing/src/main/java/io/druid/query/CacheStrategy.java

```diff
@@ -27,12 +27,11 @@
 public interface CacheStrategy<T, CacheType, QueryType extends Query<T>>
 {
   /**
-   * Computes the cache key for the given query
+   * Computes the cache key for the associated query
    *
-   * @param query the query to compute a cache key for
    * @return the cache key
    */
-  byte[] computeCacheKey(QueryType query);
+  byte[] computeCacheKey();
```

Comment:
Oh, you're right. I missed that the custom query type may exist as an extension.
I think it's better to roll back the CacheStrategy interface and make another issue to discuss about changing the interface.